### ISSUE TITLE
Research Methods for detecting road surface conditions

### ISSUE THREAD

**Issue by lfranke42** [abdefi/road-quality-classification#4]
The goal of this project is to classify the road surfaces contained in the street view images.
We need to research methods of classifying that operate automatically, fast and scalably.

### Transformer 1: 
Train a model on good streets -> Output only good streets
Then: Give the model street view images and let it generate good streets from there
Then: Diff the generated image with the original street view image

### Transformer 2:
Compress image into a few data points
Then:  Create refernce values for good/bad conditions 
Then:  Compare resulting values with reference values

### Alternative:
Use some kind of classic image recoginiton library to distinguish between different road types.
Examples: Tensorflow, OpenCV, ....

**Comment by BehrendtEric** [comment]
![Image](https://github.com/user-attachments/assets/d9491f9e-b109-401d-9531-75af83f4fd8c)

With a custom loss function for model training and a small extension on the training and validation data I reached an accuracy of 72.7% and, maybe more important, an average balanced label distance of 36,16. What the second score exactly means can be checked [train_model.py](src/model/train_model.py) file
